ios/Frame Processor/FrameHostObject.mm

```diff
@@ -78,6 +79,31 @@
     auto planesCount = CVPixelBufferGetPlaneCount(imageBuffer);
     return jsi::Value((double) planesCount);
   }
+  if (name == "depth") {
+      if (this->frame.depth != nil) {
+          jsi::Object depthFrame = jsi::Object(runtime);
+          auto toString = [this] (jsi::Runtime& runtime, const jsi::Value&, const jsi::Value*, size_t) -> jsi::Value {
+            if (this->frame.depth == nil) {
+              return jsi::String::createFromUtf8(runtime, "[closed frame]");
+            }
+            auto width = CVPixelBufferGetWidth(frame.depth);
+            auto height = CVPixelBufferGetHeight(frame.depth);
+
+            NSMutableString* string = [NSMutableString stringWithFormat:@"%lu x %lu Frame", width, height];
+            return jsi::String::createFromUtf8(runtime, string.UTF8String);
+          };
+          auto width = CVPixelBufferGetWidth(frame.depth);
+          auto height = CVPixelBufferGetHeight(frame.depth);
+          auto bytesPerRow = CVPixelBufferGetBytesPerRow(frame.depth);
+          auto planesCount = CVPixelBufferGetPlaneCount(frame.depth);
+          depthFrame.setProperty(runtime, "toString", jsi::Function::createFromHostFunction(runtime, jsi::PropNameID::forUtf8(runtime, "toString"), 0, toString));
+          depthFrame.setProperty(runtime, "width", jsi::Value((double) width));
+          depthFrame.setProperty(runtime, "height", jsi::Value((double) height));
+          depthFrame.setProperty(runtime, "bytesPerRow", jsi::Value((double) bytesPerRow));
+          depthFrame.setProperty(runtime, "planesCount", jsi::Value((double) planesCount));
+          return depthFrame;
```

Comment:
hmm, do you think it makes sense to have a separate `HostObject` for that? Then it's lazy and easier to extend.

Reply:
Yeah if we can - wasn't sure how to set it up separately and by lazy do you mean lazy access of each of the properties - what's the advantage of this in JSI, I was a little confused with how properties are accessed in the sense of `setProperty` and getting it from the object vs. `if (name == "propertyName")`?

Reply:
Well in this case it doesn't matter since you only get simple number properties, but if we had a `ArrayBuffer` of the pixels then it gets more complicated